### PR TITLE
PR: Fix some real and some potential bugs

### PR DESCRIPTION
**Pull request by novaya**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Found some real and some potential bugs

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: external-deps/spyder-kernels/spyder_kernels/customize/tests/test_umr.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def test_umr_run(user_module):
 
     from foo1.bar import square
     umr.run()
-    umr.modnames_to_reload == ['foo', 'foo.bar']
+    assert umr.modnames_to_reload == ['foo', 'foo.bar']
```
Comment (ccordoba12) [inline external-deps/spyder-kernels/spyder_kernels/customize/tests/test_umr.py:77]: Please revert this change and open a PR with it in [spyder-kernels](https://github.com/spyder-ide/spyder-kernels) with it. That's because this is a copy of that repo that we include to make contributors life easier.

File: spyder/plugins/shortcuts/widgets/summary.py [hunk 2]
```diff
@@ -80,7 +82,7 @@ def __init__(self, parent=None):
         # group shortcuts by context
         shortcuts = groupby(sorted(CONF.iter_shortcuts()), key=itemgetter(0))
 
-        for context, group_shortcuts in shortcuts:
+        for _, group_shortcuts in shortcuts:
```
Comment (ccordoba12) [inline spyder/plugins/shortcuts/widgets/summary.py:85]:
```suggestion
        for __, group_shortcuts in shortcuts:
```

This is to avoid a clash with our translations function.

**Review by ccordoba12 — COMMENTED** [review]
Thanks a lot for your help @novaya! I left a couple of small suggestions, otherwise looks good.

Please also rebase or merge with master to get the fix to our tests I made some hours ago.

**Comment by ccordoba12** [comment]
Hey @novaya, I think this is almost ready. Please merge with master to get the latest fixes to our tests and then I'll merge.

**Comment by novaya** [comment]
@ccordoba12 this is done

**Review by ccordoba12 — APPROVED** [review]
Thanks a lot for your help @novaya!
